src/dvc_data/index/fetch.py

```diff
@@ -65,6 +65,15 @@ def _collect_from_index(
                 # rely on entry.loaded
                 loaded = True
 
+            if (
+                isinstance(remote, FileStorage)
+                and remote.fs.version_aware
+                and entry.meta
+                and not entry.meta.isdir
+                and entry.meta.version_id is None
+            ):
+                entry.meta.md5 = None
+                entry.hash_info = None
```

Comment:
When we are fetching the latest version of a file from versioned file storage we should clear any existing hash information (which comes from local sources) since the latest version may have changed (this forces the hash to be re-computed in `md5()`/`save()`)

Reply:
@pmrowla Do you mean to modify index here? If so, you need to treat entries as immutables and set it into the index, as otherwise these changes won't propagate into the sqlite db underneath.

Reply:
Looks like you only need it until the DataIndexEntry creation below, right? Then use a temporary `meta` and `hash_info` vars maybe.

Reply:
Yeah it's only needed for the DataIndexEntry dict update below, it's not expected that the changes to this `entry` need to persist outside this loop scope.

I'll push an update to use temporary vars to make it clearer/more readable when I get the chance.